devel/setup-e2e-deps.sh

```diff
@@ -52,4 +54,14 @@ echo "Installing ingress-nginx into the cluster..."
 echo "Loading vault into the cluster..."
 "${SCRIPT_ROOT}/addon/vault/install.sh" &
 
-wait
+# Wait for all background jobs to finish and exit with non-zero if any of them fail
+# See https://stackoverflow.com/a/515170/919436
+for job in $(jobs -p); do
+    wait $job || let "EXIT+=1"
+done
+
+if [[ "$EXIT" > 0 ]]; then
+    echo "ERROR: ${EXIT} setup jobs failed. Check logs above for details."
+fi
+
+exit $EXIT
```

Comment:
The sample-webhook wasn't getting deployed on K8S < 1.15 because it included v1alpha2 resources which are no longer defined by the legacy CRD yamls.

But this script exited with `0` because bare `wait` *always* exits with 0.

With this change, the script should exit with a non-zero code if any of the background jobs fails, and fail the e2e test early, with a more useful error message.

It could be improved further though.

Reply:
Good catch!